cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java

```diff
@@ -172,9 +172,17 @@ static void applyTemplate(String orgName, String templatePkgName, String version
         Path balaPath = balaCache.resolve(
                 ProjectUtils.getRelativeBalaPath(orgName, templatePkgName, version, null));
         //First we will check for a bala that match any platform
-        String platform = findPlatform(balaPath);
-        balaPath = balaCache.resolve(
-                ProjectUtils.getRelativeBalaPath(orgName, templatePkgName, version, platform));
+        String platform = ANY_PLATFORM;
+        if (!Files.exists(balaPath)) {
+            for (JvmTarget supportedPlatform : JvmTarget.values()) {
```

Comment:
Shall we move this logic to seperate fucntion and replace two places. We can have function name like getBaloPathforJavaPlatform